### DISCUSSION TITLE
ESP32-S3 build error undefined reference to `usocket_events_deinit'

### DISCUSSION THREAD

**Discussion by git-n-pissed** [micropython/micropython#10192] [ESP32] [answered]
ESP-IDF 4.4 Commit: f90c12fc3cc5abb47a7f1088ada43a21b31d0357 (12/02/2022)
MicroPython Commit: ac5934c96b66555ba8fe43465b34cb2b2a692ff0 (11/17/2022)
Board: GENERIC_S3

Commands:
```
source ../esp-idf/export.sh
make -j8 v=1 -C ./micropython/mpy-cross
make -j8 submodules v=1 BOARD=GENERIC_S3 -C ./micropython/ports/esp32
make -j8 v=1 BOARD=GENERIC_S3 -C ./micropython/ports/esp32
```

My first attempts at building MicroPython for an ESP32-S3 is resulting in the following errors when building.  All the commands except the last one seem to execute without error.  Some (seemingly) superfluous output has removed:

```
[1438/1440] Linking CXX executable micropython.elf
FAILED: micropython.elf
...
/home/gitnpissed/.espressif/tools/xtensa-esp32s3-elf/esp-2021r2-patch5-8.4.0/xtensa-esp32s3-elf/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld: esp-idf/main/libmain.a(main.c.obj):(.literal.mp_task+0x30): undefined reference to `usocket_events_deinit'
/home/gitnpissed/.espressif/tools/xtensa-esp32s3-elf/esp-2021r2-patch5-8.4.0/xtensa-esp32s3-elf/bin/../lib/gcc/xtensa-esp32s3-elf/8.4.0/../../../../xtensa-esp32s3-elf/bin/ld: esp-idf/main/libmain.a(main.c.obj): in function `mp_task':
/mnt/c/Git/micropython/ports/esp32/main.c:178: undefined reference to `usocket_events_deinit'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
ninja failed with exit code 1
-e See https://github.com/micropython/micropython/wiki/Build-Troubleshooting
make: *** [Makefile:40: all] Error 1
make: Leaving directory '/mnt/c/Git/micropython/ports/esp32'
```

main.c does have a reference to `usocket_events_deinit`, but it is on line 217, not 178 as indicated by the output:

```
    // deinitialise peripherals
    machine_pwm_deinit_all();
    // TODO: machine_rmt_deinit_all();
    machine_pins_deinit();
    machine_deinit();
    usocket_events_deinit();

    mp_deinit();
    fflush(stdout);
    goto soft_reset;
```

And `usocket_events_deinit` is declared in `modnetwork.h` which is included in `main.c`, so I'm not sure where the disconnect is.

Any guidance would be appreciated.

**Comment by ThinkTransit** [comment, marked as answer]
What commands are you running to build and have you built the submodules?

**Reply by git-n-pissed** [reply]
> What commands are you running to build and have you built the submodules?

Yes I have built the submodules.  Updated my OP with the commands being used.  All the commands except the last one seem to execute without error.

**Reply by git-n-pissed** [reply]
I've found my error is being caused by having tried to disable a few things in `mpconfigport.h`, namely `MICROPY_PY_WEBREPL`.  Maybe I've misunderstood the purpose of the file, but in the past I had used it to disable certain software features I didn't want in my build, but attempting to disable `MICROPY_PY_WEBREPL` like I did with the ESP8266 isn't working.

**Reply by jimmo** [reply]
Yeah I think this is a bug in esp32/main.c -- It should only call `usocket_events_deinit()` with a `#if MICROPY_PY_USOCKET_EVENTS` around it.

**Comment by git-n-pissed** [comment]
Fixed in commit https://github.com/micropython/micropython/commit/9bec52a2f8371847e1d75fee84bf37a9b2876cd4

Thanks @jimmo
